core/utils/dmljob.py

```diff
@@ -62,7 +65,7 @@ def serialize_job(dmljob_obj):
     return {
         'weights': serialize_weights(dmljob_obj.weights),
         'job_data': rest
-    }
+    } if dmljob_obj.weights else {'job_data': rest}
```

Comment:
When is `dmljob_obj.weights` won't be defined? `communicate` and possibly `average` type of jobs? Please confirm the intention of this check and if this is the case, we may need to create new types of (job) classes that the Scheduler and Runner can take in the upcoming PRs.

Reply:
yes

Reply:
no, 'initialize' jobs.

Reply:
Ok.